ariadne_codegen/plugins/shorter_results.py

```diff
@@ -0,0 +1,179 @@
+)
+
+
+class ShorterResultsPlugin(Plugin):
+    """
+    Make single field return types expanded.
+
+    All client method that returns type with a single field will be expanded and
+    instead return the type of the inner field.
+    """
+
+    def generate_result_class(
+        self,
+        class_def: ast.ClassDef,
+        operation_definition: OperationDefinitionNode,
+        selection_set: SelectionSetNode,
+    ) -> ast.ClassDef:
+        """Store a map of all classes and its ast"""
+        self.class_dict[class_def.name] = class_def
+
+        return super().generate_result_class(
+            class_def, operation_definition, selection_set
+        )
+
+    def generate_client_module(self, module: ast.Module) -> ast.Module:
+        """Add extra import needed after expanding the inner field"""
+        if len(self.extended_imports) == 0:
+            return super().generate_client_module(module)
+
+        for stmt in module.body:
+            # We know we've already imported the wrapping type so we're only
+            # interested in `ImportFrom`.
+            if not isinstance(stmt, ast.ImportFrom):
+                continue
+
+            # The module we import from is always the same as the method we
+            # modified when changing the return type.
+            if stmt.module not in self.extended_imports:
+                continue
+
+            # Add all additional imports discovered when generating the client
+            # method.
+            for additional_import in self.extended_imports[stmt.module]:
+                stmt.names.append(ast.alias(name=additional_import))
+
+        return super().generate_client_module(module)
+
+    def generate_client_method(
+        self, method_def: Union[ast.FunctionDef, ast.AsyncFunctionDef]
+    ) -> Union[ast.FunctionDef, ast.AsyncFunctionDef]:
+        """
+        Change the method generated in the client to call the inner field and
+        change the return type.
+        """
+        if len(method_def.body) < 1:
+            return super().generate_client_method(method_def)
+
+        # TODO: Can we assume that the return statement is always the last
+        # statement?
+        return_stmt = method_def.body[-1]
+        if not isinstance(return_stmt, ast.Return):
+            return super().generate_client_method(method_def)
+
+        # Get the originally returned class so we can look at the class
+        # definition to see how many fields it has.
+        if not isinstance(method_def.returns, ast.Name):
+            return super().generate_client_method(method_def)
+
+        return_class = method_def.returns.id
+        if return_class not in self.class_dict:
+            return super().generate_client_method(method_def)
+
+        return_class_ast = self.class_dict[return_class]
+
+        # If we're not a single method data class move on!
+        if len(return_class_ast.body) != 1:
+            return super().generate_client_method(method_def)
```

Comment:
Right... I didn't address this at all yet, need to dig some more. Do you have any suggestions or ideas on how to tackle this?

Reply:
Ok I did a quick pass just to get an idea of this. The first case is straight forward, if we only have a single field, check that we inherit from `BaseModel` (I guess there are no other classes to consider?). If we inherit from anything else we don't unwrap the field. We could traverse the inherited class (if we somehow get the `ClassDef`) but I can't think of a scenario where we would inherit from a class _without_ fields.

For the second example that's a bit trickier because we never see `TestQuery` in `generate_result_class` so it won't get added to our dict and thus I won't be able to get what class it inherits from. I see we get the code for the class in `generate_result_types_code` but that one is called _after_ `generate_client_method` so even if I parse the code to get all the generated classes it will be too late. Also we don't have the ast for the fragment either but that's exposed in `generate_fragments_module` which is also called _after_ `generate_client_method` so not sure how much that will help.

Reply:
Sorry, I missed this comment, I addressed this issue [here](https://github.com/mirumee/ariadne-codegen/pull/141#discussion_r1193909475)